src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -755,6 +755,27 @@ public int ConnectionIdleLifetime
         }
         int _connectionIdleLifetime;
 
+        /// <summary>
+        /// The time to wait before force closing connections in the pool if the count
+        /// of all connections exceeds MinPoolSize.
+        /// </summary>
+        /// <value>The time (in seconds) to wait. The default value is 0 (disabled).</value>
+        [Category("Pooling")]
+        [Description("The time to wait before force closing connections in the pool if the count of all connections exceeds MinPoolSize.")]
+        [DisplayName("Connection Max Lifetime")]
+        [NpgsqlConnectionStringProperty]
+        [DefaultValue(0)]
+        public int ConnectionMaxLifetime
```

Comment:
If we want to align with SQL Client, we should consider calling this `Connection Lifetime` (see [this doc page](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnection.connectionstring?view=netframework-4.7.2)). This parameter actually existed before 3.1, but actually meant what `Connection Idle Lifetime` means (it was renamed, see [the release notes](http://www.npgsql.org/doc/release-notes/3.1.html). 4.1 seems like the right time to add the "max lifetime" feature with the SQL Client name, for consistency.